src/screens/PostsDisplayScreen/PostCard/PostCard.tsx

```diff
@@ -0,0 +1,374 @@
+import {useDispatch, useSelector} from "react-redux";
+import React, {useEffect, useState} from "react";
+import axios from "axios";
+import {URI} from "../../../URI";
+import {TouchableOpacity, View, Image, TouchableWithoutFeedback, Text, Modal, StyleSheet} from "react-native";
+import {addLikes, removeLikes} from "../../../redux/actions/postActions";
+import PostDetailsCard from "./PostDetailsCard";
+import getTimeDuration from "./timeGen";
+import {useNavigation} from "@react-navigation/native";
+import {Post, User} from "../../../redux/reducers/User";
+import {getPosts} from "../../../redux/actions/getPosts";
+
+type Props = {
+    item: Post;
+    isReply?: boolean | null;
+    postId?: string | null;
+    replies?: boolean | null;
+};
+
+export const PostCard = ({item, isReply, postId, replies}: Props) => {
+    const navigation = useNavigation();
+    const {user, token, users} = useSelector((state: any) => state.user);
+    const {posts} = useSelector((state: any) => state.post);
+    const [openModal, setOpenModal] = useState(false);
+    const dispatch = useDispatch();
+    const [userInfo, setUserInfo] = useState({
+        name: '',
+        avatar: {
+            url: 'https://res.cloudinary.com/dshp9jnuy/image/upload/v1665822253/avatars/nrxsg8sd9iy10bbsoenn.png',
+        },
+    });
+
+    const time = item?.createdAt;
+    const formattedDuration = getTimeDuration(time!);
+
+    const profileHandler = async (e: User) => {
+        await axios
+            .get(`${URI}/get-user/${e._id}`, {
+                headers: {Authorization: `Bearer ${token}`},
+            })
+            .then(res => {
+                if (res.data.user._id !== user._id) {
+                    navigation.navigate('UserProfileScreen', {
+                        id: res.data.user._id,
+                    });
+                } else {
+                    navigation.navigate('ProfileScreen' as never);
+                }
+            });
+    };
+
+    const reactsHandler = (e: User) => {
+        if (item?.likes?.length !== 0) {
+            const isLikedBefore = item?.likes?.find((i: any) => i.userId === user._id);
+            if (isLikedBefore) {
+                removeLikes({postId: postId ? postId : e._id, posts, user})(dispatch);
+            } else {
+                addLikes({postId: postId ? postId : e._id, posts, user})(dispatch);
+            }
+        } else {
+            addLikes({postId: postId ? postId : e._id, posts, user})(dispatch);
+        }
```

Comment:
It's too much logic for UI component. Let's move it to a thunk.
```ts
dispatch(reactToPost(postId))
```
And then inside the `reactToPost` thunk we will decide what action to dispatch